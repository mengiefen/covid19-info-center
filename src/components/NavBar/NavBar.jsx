import React from 'react';
import { useLocation, useNavigate } from 'react-router-dom';
import { HiChevronLeft, HiOutlineMicrophone } from 'react-icons/hi';
import { FiSettings } from 'react-icons/fi';
import Nav from './NavBar.style';

const NavBar = () => {
  const navigate = useNavigate();
  const params = useLocation();

  const url = params.pathname.split('/').slice(1, 3);

  const region = url[1]
    ? url[1].split('%20').join(' ')
    : url[0].split('%20').join(' ');

  return (
    <Nav>
      {params.pathname !== '/' && (
        <div className="navigation-tab" url={url[0]}>
          <HiChevronLeft className="arrow-left" onClick={() => navigate(-1)} />
        </div>
      )}
      <h2 className="header-title">{region}</h2>
      {url[0] === '' && (
        <h2 className="logo" onClickCapture={() => navigate('/')}>
          COVID
          <span>Info</span>
        </h2>
      )}
      <div className="settings-tab">
        <HiOutlineMicrophone className="btn-setting" />
        <FiSettings className="btn-microphone" />
      </div>
    </Nav>
  );
};

export default NavBar;
