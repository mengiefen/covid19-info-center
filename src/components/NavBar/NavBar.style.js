import styled from 'styled-components';

const Nav = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0.25rem 0.75rem;
  box-shadow: 1px 0px 5px gray;
  margin-bottom: 0.5rem;
  z-index: 10;

  @media screen and (min-width: 768px) {
    height: 7.5vh;
  }

  button {
    display: flex;
    align-items: center;
    background-color: transparent;
    color: ${({ theme }) => theme.colors.text};
    border: 1px solid transparent;
    font-size: 1rem;
    border-radius: 20px;
    padding: 0 0.5rem 0 0;
    transition: all 0.5s ease;
    margin-right: 0.5rem;

    &:hover {
      border: 1px solid #9ae5e6;
      background-color: ${({ theme }) => theme.colors.dark};
    }
  }

  .arrow-left {
    display: ${(props) => (props.url === '' ? 'none' : 'block')};
    font-size: 2rem;
    cursor: pointer;
    padding: 0.2rem;
    transition: all 0.5s;
  }

  .logo {
    color: #fa9900;
    cursor: pointer;
    font-weight: 500;
    font-size: 1.4rem;
    letter-spacing: 1.2;
    text-transform: uppercase;

    @media screen and (min-width: 768px) {
      letter-spacing: 2;
      font-size: 1.8rem;
    }
  }

  .logo span {
    color: ${({ theme }) => theme.colors.text};
  }

  .header-title {
    color: ${({ theme }) => theme.colors.text};
    font-size: 1rem;
    max-width: 60%;
    text-transform: uppercase;
    letter-spacing: 1.2px;
    max-width: 60%;    
    overflow: hidden;
    text-overflow: ellipsis;
  }

  .search-icon {
    font-size: 2.5rem;
    color: ${({ theme }) => theme.colors.text};
    cursor: pointer;
    padding: 0.5rem;
    border-radius: 50%;
    transition: all 0.5s;

    &:hover {
      border: 1px solid #9ae5e6;
      background-color: ${({ theme }) => theme.colors.cyan};
    }
  }

  .settings-tab {
    display: flex;
    margin-left: 0.5rem;
    .btn-setting,
    .btn-microphone,
    .btn-home {
      background-color: transparent;
      font-size: 2.5rem;
      color: ${({ theme }) => theme.colors.text};
      padding: 0.5rem;
      border-radius: 50%;
      cursor: pointer;
      transition: all 0.5s ease-out;

      &:hover {
        border: 1px solid #9ae5e6;
        background-color: ${({ theme }) => theme.colors.cyan};
      }
    }
  }
`;

export default Nav;
