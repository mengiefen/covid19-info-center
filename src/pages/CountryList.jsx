import React, { useEffect, useState } from 'react';
import { HiOutlineArrowCircleRight, HiOutlineSearch } from 'react-icons/hi';
import { useNavigate, useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { RegionCard, CardWrapper, Container } from './Details.styled';
import AFRICA from '../assets/images/continents/africa1.svg';
import EUROPE from '../assets/images/continents/europe.svg';
import NORTH_AMERICA from '../assets/images/continents/north_america.svg';
import SOUTH_AMERICA from '../assets/images/continents/latin_america.svg';
import AUSTRALIA from '../assets/images/continents/australia.svg';
import ASIA from '../assets/images/continents/asia2.svg';
import { getCovidDataFromAPI } from '../redux/country/covidDataReducer';
import { getAllInfoFromAPI } from '../redux/country/countryInfoReducer';

const CountryList = () => {
  const params = useParams();
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const covidData = useSelector((state) => state.covidData);
  const countryInfo = useSelector((state) => state.countryInfo);
  const [countries, setCountries] = useState([]);
  const [merged, setMerged] = useState([]);
  const [searchQuery, setSearchQuery] = useState('');

  const continentMap = {
    Africa: AFRICA,
    Europe: EUROPE,
    Australia: AUSTRALIA,
    Asia: ASIA,
    'South America': SOUTH_AMERICA,
    'North America': NORTH_AMERICA,
  };

  useEffect(() => {
    const newMerged = countryInfo.map((info) => {
      const value = covidData.find((country) => country.name === info.name);
      return { ...info, ...value };
    });
    setMerged(newMerged);
    setCountries(newMerged);
  }, []);

  useEffect(() => {
    if (covidData.length === 0 || countryInfo.length === 0) {
      dispatch(getCovidDataFromAPI());
      dispatch(getAllInfoFromAPI());
    }
  }, []);

  const handleNavigation = (id) => {
    navigate(`/${params.continent}/${id}`);
  };

  const handleSearchFilter = (searchValue) => {
    setSearchQuery(searchValue);
    if (searchQuery) {
      const filteredData = merged.filter((country) => {
        const newValue = country.name
          .toLowerCase()
          .includes(searchQuery.toLowerCase());
        return newValue;
      });
      setCountries(filteredData);
    } else {
      setCountries(merged);
    }
  };

  return (
    <Container bg={continentMap[params.continent]}>
      <form onSubmit={(e) => e.preventDefault()}>
        <input
          type="text"
          id="country"
          className="country-search"
          placeholder="Search by country"
          onChange={(e) => handleSearchFilter(e.target.value)}
        />
        <button type="button" className="btn-search">
          <HiOutlineSearch />
        </button>
      </form>
      <CardWrapper>
        {countries
          .filter(
            (country) => country.continent
            && country.continent
              .toLowerCase()
              .includes(params.continent.toLowerCase()),
          )
          .map(
            (countryData) => countryData.todayConfirmed && (
              <RegionCard key={countryData.id}>
                <h3 className="country-name">{countryData.name}</h3>
                <HiOutlineArrowCircleRight
                  onClick={() => handleNavigation(countryData.name)}
                  className="open-icon"
                />
                <div className="summary">
                  <div className="col-1">
                    <h3>Active</h3>
                    <h4>{`Total: ${countryData.todayConfirmed}`}</h4>
                    <h4>{`Last 24: ${countryData.todayNewConfirmed}`}</h4>
                  </div>
                  <div className="col-2">
                    <h3>Deaths</h3>
                    <h4>{`Total: ${countryData.todayDeaths}`}</h4>
                    <h4>{`Last 24: ${countryData.todayNewDeaths}`}</h4>
                  </div>
                  <div className="col-3">
                    <h3>Recovered</h3>
                    <h4>{`Total: ${countryData.todayRecovered}`}</h4>
                    <h4>{`Last 24: ${countryData.todayRecovered}`}</h4>
                  </div>
                </div>
              </RegionCard>
            ),
          )}
      </CardWrapper>
    </Container>
  );
};

export default CountryList;
