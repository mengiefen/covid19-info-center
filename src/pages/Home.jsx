import React from 'react';
import { useNavigate } from 'react-router-dom';
import { HiOutlineArrowCircleRight } from 'react-icons/hi';
import { Container, Card, CardWrapper } from './Home.styled';
import AFRICA from '../assets/images/continents/africa1.svg';
import ASIA from '../assets/images/continents/asia.svg';
import AUSTRALIA from '../assets/images/continents/asia2.svg';
import EUROPE from '../assets/images/continents/europe.svg';
import NORTH from '../assets/images/continents/north_america.svg';
import LATIN from '../assets/images/continents/latin_america.svg';

const Home = () => {
  const navigate = useNavigate();
  const continets = [
    { id: 1, continent: 'Africa', img: AFRICA },
    { id: 2, continent: 'Asia', img: ASIA },
    { id: 3, continent: 'Australia', img: AUSTRALIA },
    { id: 4, continent: 'Europe', img: EUROPE },
    { id: 5, continent: 'North America', img: NORTH },
    { id: 6, continent: 'South America', img: LATIN },
  ];
  return (
    <Container>
      <CardWrapper>
        {continets.map((name) => (
          <Card
            key={name.id}
            bgImg={name.img}
            onClick={() => {
              navigate(`/${name.continent}`);
            }}
          >
            <h3 className="continent-name">{name.continent}</h3>
            <HiOutlineArrowCircleRight
              onClick={() => {
                navigate(`/${name.continent}`);
              }}
              className="open-icon"
            />
          </Card>
        ))}
      </CardWrapper>
    </Container>
  );
};

export default Home;
