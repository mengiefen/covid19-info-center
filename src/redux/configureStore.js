import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import countryInfoReducer from './country/countryInfoReducer';
import covidDataReducer from './country/covidDataReducer';

const rootReducer = combineReducers({
  countryInfo: countryInfoReducer,
  covidData: covidDataReducer,
});

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);
const store = createStore(persistedReducer, applyMiddleware(thunk, logger));
const persistor = persistStore(store);

export { persistor, store };
