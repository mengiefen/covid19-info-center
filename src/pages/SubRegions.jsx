import React from 'react';
import { useParams } from 'react-router-dom';
import { Container } from './Home.styled';

const SubRegions = () => {
  const param = useParams();
  return <Container>{`SOME THING ${param.subRegions}`}</Container>;
};

export default SubRegions;
