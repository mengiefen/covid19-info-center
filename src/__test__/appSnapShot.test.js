import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import { store } from '../redux/configureStore';
import NavBar from '../components/NavBar/NavBar';
import Home from '../pages/Home';
import SubRegions from '../pages/SubRegions';
import CountryList from '../pages/CountryList';

const theme = {
  colors: {
    bg: '#2f4454',
    shadow: '#c8c8c8',
    text: '#fefefe',
    coffee: '#2e151b',
    pink: '#da7b93',
    cyan: '#376e6f',
    semiCyan: '#376e6faa',
    dark: '#1c3334',
    semiDark: '#1c333444',
  },

  devices: {
    mobile: '567px',
    note: '767px',
    tab: '1024px',
    desktop: '1280px',
  },
};

it('Home Page renders correctly when there are no items', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Router>
            <Home />
          </Router>
        </ThemeProvider>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Nav Bar renders correctly when there are no items', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Router>
            <NavBar />
          </Router>
        </ThemeProvider>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Sub Regions Page renders correctly when there are no items', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Router>
            <SubRegions />
          </Router>
        </ThemeProvider>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Countrly List renders correctly when there are no items', async () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Router>
            <CountryList />
          </Router>
        </ThemeProvider>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
