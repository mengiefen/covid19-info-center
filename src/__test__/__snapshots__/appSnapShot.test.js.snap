// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countrly List renders correctly when there are no items 1`] = `
<section
  className="sc-jSMfEi drbRiE"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="country-search"
      id="country"
      onChange={[Function]}
      placeholder="Search by country"
      type="text"
    />
    <button
      className="btn-search"
      type="button"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </form>
  <div
    className="sc-iBkjds eWCXSa"
  />
</section>
`;

exports[`Home Page renders correctly when there are no items 1`] = `
<section
  className="sc-gsnTZi hjRMeY"
>
  <div
    className="sc-dkzDqf gpYKFn"
  >
    <div
      className="sc-hKMtZM inmPkC"
      onClick={[Function]}
    >
      <h3
        className="continent-name"
      >
        Africa
      </h3>
      <svg
        className="open-icon"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="sc-hKMtZM kwfDHD"
      onClick={[Function]}
    >
      <h3
        className="continent-name"
      >
        Asia
      </h3>
      <svg
        className="open-icon"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="sc-hKMtZM irncPF"
      onClick={[Function]}
    >
      <h3
        className="continent-name"
      >
        Australia
      </h3>
      <svg
        className="open-icon"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="sc-hKMtZM iaipOZ"
      onClick={[Function]}
    >
      <h3
        className="continent-name"
      >
        Europe
      </h3>
      <svg
        className="open-icon"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="sc-hKMtZM coMoYZ"
      onClick={[Function]}
    >
      <h3
        className="continent-name"
      >
        North America
      </h3>
      <svg
        className="open-icon"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="sc-hKMtZM hbZZXg"
      onClick={[Function]}
    >
      <h3
        className="continent-name"
      >
        South America
      </h3>
      <svg
        className="open-icon"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </div>
  </div>
</section>
`;

exports[`Nav Bar renders correctly when there are no items 1`] = `
<header
  className="sc-bczRLJ bfFjox"
>
  <h2
    className="header-title"
  />
  <h2
    className="logo"
    onClickCapture={[Function]}
  >
    COVID
    <span>
      Info
    </span>
  </h2>
</header>
`;

exports[`Sub Regions Page renders correctly when there are no items 1`] = `
<section
  className="sc-gsnTZi hjRMeY"
>
  SOME THING undefined
</section>
`;
